internal/distro/rhel9/ami.go

```diff
@@ -491,38 +403,95 @@ func rhelEc2SapPackageSet(t *imageType) rpmmd.PackageSet {
 	}.Append(rhelEc2CommonPackageSet(t)).Append(SapPackageSet(t))
 }
 
-func mkAMIImgTypeX86_64(osVersion string) imageType {
-	it := amiImgTypeX86_64
-	it.defaultImageConfig = defaultAMIImageConfigX86_64(osVersion)
+func mkEc2ImgTypeX86_64(osVersion string, rhsm bool) imageType {
+	it := ec2ImgTypeX86_64
+	ic := defaultEc2ImageConfigX86_64(osVersion, rhsm)
+	it.defaultImageConfig = ic
 	return it
 }
 
-func mkEC2SapImgTypeX86_64(osVersion string) imageType {
-	it := ec2SapImgTypeX86_64
-	it.defaultImageConfig = sapImageConfig(osVersion).InheritFrom(defaultEc2ImageConfigX86_64(osVersion))
+func mkAMIImgTypeX86_64(osVersion string, rhsm bool) imageType {
+	it := amiImgTypeX86_64
+	ic := defaultAMIImageConfigX86_64(osVersion, rhsm)
+	if rhsm && common.VersionLessThan(osVersion, "9.1") {
+		ic = appendRHSM(ic)
```

Comment:
This should have been removed as well, since it is already done in `defaultAMIImageConfig()` called from `defaultAMIImageConfigX86_64()`. Also the configuration is applied unconditionally to AMIs regardless of the minor version so the condition in `defaultAMIImageConfig()` is actually the correct one. 

Nevertheless this "bug" has no effect on resulting functionality.

Reply:
I'll clean this up in a followup.